src/components/MapView/Layers/raster-utils.ts

```diff
@@ -120,21 +121,60 @@ export function getWCSUrl(
   return formatUrl(baseUrl, params);
 }
 
+export function getWCS2Url(
+  layer: WMSLayerProps,
+  date: string | undefined,
+  extent: Extent,
+) {
+  const params = {
+    service: 'WCS',
+    request: 'GetCoverage',
+    version: layer.wcsConfig?.version,
+    coverageId: layer.serverLayerName,
+  };
+
+  // Subsets are used as spatial and temporal filters.
+  // For mote info: https://docs.geoserver.geo-solutions.it/edu/en/wcs/get.html
+  const spatialSubsets: string[] = [
+    `Long(${extent[0]}, ${extent[2]})`,
+    `Lat(${extent[1]}, ${extent[3]})`,
+  ];
+
+  const subsets = date
+    ? [...spatialSubsets, `time("${date}")`]
+    : spatialSubsets;
+
+  const formattedUrl = formatUrl(layer.baseUrl, params);
+
+  const formattedSubsets = subsets.map(s => `subset=${s}`).join('&');
+
+  return `${formattedUrl}&${formattedSubsets}`;
+}
+
 export function WCSRequestUrl(
-  baseUrl: string,
-  layerName: string,
+  layer: WMSLayerProps,
   date: string | undefined,
   extent: Extent,
-  resolution = 256,
   maxPixels = 5096,
 ) {
+  const { baseUrl, serverLayerName, wcsConfig } = layer;
   const [minX, minY, maxX, maxY] = extent;
+
   if (minX > maxX || minY > maxY) {
     throw new Error(
-      `Could not generate WCS request for ${baseUrl}/${layerName}: the extent ${extent} seems malformed or else may contain "wrapping" which is not implemented in the function 'WCSRequestUrl'`,
+      `Could not generate WCS request for ${baseUrl}/${serverLayerName}: the extent ${extent} seems malformed or else may contain "wrapping" which is not implemented in the function 'WCSRequestUrl'`,
     );
   }
 
+  if (
+    wcsConfig?.version &&
+    wcsConfig?.version !== WcsGetCoverageVersion.oneZeroZero
+  ) {
+    return getWCS2Url(layer, date, extent);
+  }
```

Comment:
It might be simpler to just do:


if (wcsConfig?.version === WcsGetCoverageVersion.twoZeroZero){
  return getWCS2Url(layer, date, extent);
}

Reply:
Included in cd36ad92b0dbf74b19a44bf10501f7751b9b6560